package/MDAnalysis/coordinates/memory.py

```
@@ -480,3 +480,33 @@ def __repr__(self):
                     nframes=self.n_frames,
                     natoms=self.n_atoms
                 ))
+    
+    def add_transformations(self, transform_list):
+        """ Adds all the transformations to be applied to the trajectory.
+        Overrides :meth:`~MDAnalysis.coordinates.base.ProtoReader.add_transformations`
+        to avoid unintended behaviour where the coordinates of each frame are transformed
+        multiple times when iterating over the trajectory.
+        
+        In this method, the trajectory is modified all at once and once only.
+        """
+        try:
+            self.transformations = transform_list
+        except:
```

There is an other bare except here.